src/main/java/com/uber/profiling/YamlConfigProvider.java

```diff
@@ -158,18 +157,33 @@ public void setFilePath(String filePath) {
     
     private static void addConfig(Map<String, Map<String, List<String>>> config, String override, String key, Object value) {
         Map<String, List<String>> configMap = config.computeIfAbsent(override, k -> new HashMap<>());
-        List<String> configValueList = configMap.computeIfAbsent(key, k -> new ArrayList<>());
         
         if (value instanceof List) {
+            int count = 0;
             for (Object entry : (List)value) {
-                configValueList.add(entry.toString());
+                configMap.put(key + "[" + count++ + "]", Arrays.asList(entry.toString()));
```

Comment:
We do not want to put the index bracket into config key name (e.g. key[0]=value1, key[1]=value2). This will break our existing code, since we assume the config is: key=[value1, value2].